src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPServerSideOnTypeFormattingFixtureTestCase.java

```diff
@@ -62,11 +68,14 @@ protected void setTriggerCharacters(@NotNull String... triggerCharacters) {
      * @param fileBodyBefore           the file body before including the embedded typing imperative comment
      * @param fileBodyAfter            the file body after the character has been typed and formatting applied
      * @param mockOnTypeFormattingJson the mock on-type formatting JSON response
+     * @param clientConfigCustomizer   a function that allows the caller to customize the client configuration as needed
+     *                                 for the test scenario
      */
     protected void assertOnTypeFormatting(@NotNull String fileName,
                                           @NotNull String fileBodyBefore,
                                           @NotNull String fileBodyAfter,
-                                          @NotNull String mockOnTypeFormattingJson) {
+                                          @NotNull String mockOnTypeFormattingJson,
+                                          @Nullable Consumer<ClientConfigurationSettings> clientConfigCustomizer) {
```

Comment:
Which meant that I had to add the ability to customize client config for each test.